src/query/query_eval.rs

```diff
@@ -30,6 +30,19 @@ fn eval_latest<'a>(
     }
 }
 
+fn packet_has_param(packet: &Packet, key: &str, value: &str) -> bool {
+    match &packet.parameters {
+        Some(params) => {
+            if let Some(existing_value) = params.get(key) {
+                *existing_value == value
```

Comment:
Not sure this asterisk is needed?

Reply:
Thanks! It is not, I've removed that now